src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,109 @@
+import { FC, useState, useEffect } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from "../../api";
+
+export const PeoplePage: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const { personSlug } = useParams();
+
+  const personInfo = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const personName = (name: string | null) => {
+    return peoples.find(person => person.name === name);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(setPeoples)
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, []);
+
+  if (loading) {
+    return <Loader />;
+  }
+
+  if (error) {
+    return (
+      <p data-cy="peopleLoadingError" className="has-text-danger">
+        Something went wrong
+      </p>
+    );
+  }
+
+  if (!peoples.length) {
+    return <p data-cy="noPeopleMessage">There are no people on the server</p>;
+  }
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+        <div className="block">
+          <div className="box table-container">
+          <table
+            data-cy="peopleTable"
+            className="table is-striped is-hoverable is-narrow is-fullwidth"
+          >
+          <thead>
+            <tr>
+              {personInfo.map(info => (
+                <th key={info}>{info}</th>
+              ))}
+            </tr>
+          </thead>
+
+          <tbody>
+            {peoples.map(person => {
```

Comment:
You are calling `personName` function for each person in the `peoples` array. This could be inefficient if the array is large. Consider creating a map of people by name before rendering, and then just look up the mother and father by name from the map.